Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -321,7 +331,9 @@ public struct TLSConfiguration {
                  keyLogCallback: NIOSSLKeyLogCallback?,
                  renegotiationSupport: NIORenegotiationSupport,
                  additionalTrustRoots: [NIOSSLAdditionalTrustRoots],
-                 sendCANameList: Bool = false) {
+                 sendCANameList: Bool = false,
+                 pskCallback: PSKIdentityCallbackManager? = nil,
+                 client: Bool = false) {
```

Comment:
What is the `client` argument for?

Reply:
The `client` argument is being used to delineate between how the `NIOSSLContext` sets the callback for the particular context, e.g., `CNIOBoringSSL_SSL_CTX_set_psk_client_callback` or `CNIOBoringSSL_SSL_CTX_set_psk_server_callback`.

Now, above you mentioned:
> I think we need two callbacks, not one. The server is supposed to have access to the identity asserted by the peer so it can choose. Additionally, the client should be given an identity hint.

If you wanted to go in the direction of adding a client or server based callback to `TLSConfiguration` e.g., `NIOPSKServerCallback` and `NIOPSKClientCallback`, then this `client` flag can go away entirely.

Thoughts @Lukasa ?

Reply:
I still have some work left on this PR, but I split up the callbacks into a client and server side callback and put them both under `PSKIdentityCallbackManager` in [7742aa0](https://github.com/apple/swift-nio-ssl/pull/369/commits/7742aa0cc15f7b191af71578fd2227c7041a7cba).  This removes the `client` flag also.  Wanted to get your thoughts on this @Lukasa ?